### DISCUSSION TITLE
[Engine] My Documentation is not Build

### DISCUSSION THREAD

**Discussion by Blackjacx** [SwiftPackageIndex/SwiftPackageIndex-Server#2007] [Q&A] [answered]
Hey there, 

first of all thanks for building this awesome index! I love it and it's super helpful 👍

Now to the question. It seems that the documentation as described [here](https://blog.swiftpackageindex.com/posts/auto-generating-auto-hosting-and-auto-updating-docc-documentation) is not built for my project [Engine](https://swiftpackageindex.com/Blackjacx/Engine). I added the [.spi.yml](https://github.com/Blackjacx/Engine/blob/develop/.spi.yml) but do not have the [DocC plugin](https://github.com/tbointeractive/bytes/blob/9444dab5f7e3c40c099ea34beb0fa4761dfb908a/Package.swift#L32-L37) in my Package.swift. I'm unsure what I'm doing wrong. 

Also my [build output](https://swiftpackageindex.com/builds/43351E55-F6E7-4CBB-8A7A-FCCAC9E62E93) seems to be different from e.g. the [bytes project](https://swiftpackageindex.com/builds/F0F7FE7D-B3D4-4E26-84E1-FC6CFC2F3E53) which shows the documentation link:

![Screen Shot 2022-09-05 at 10 10 49](https://user-images.githubusercontent.com/794372/188400783-696080d1-e946-47d2-9fde-4ff249365fe2.png)

**Comment by finestructure** [comment, marked as answer]
Hi @Blackjacx , the problem is that you've specified a platform but got the name slightly wrong :)

We're defaulting to build for macOS, so you don't actually have to specify it (and if you want to specify it explicitly, it needs to be `macos-spm` or `macos-xcodebuild`).

I'll make a change so we default `macos` to `macos-spm` – we don't actually need to be that strict when decoding this value.

I've also raised a PR to fix this: https://github.com/Blackjacx/Engine/pull/8

**Reply by finestructure** [reply]
Tracked here: https://github.com/SwiftPackageIndex/SPIManifest/issues/11

**Reply by Blackjacx** [reply]
Oh nice thanks a lot! I think I had it like that before, but maybe the version was wrong. Will the runner pickup the latest commit (merge of your PR) to build the documentation or the version tag specified in `spi.yml`? If the latter is true I would have to make a new release.

**Reply by finestructure** [reply]
We rebuild default branch changes automatically but coalesce over 24h. I.e. it can take up to 24h for default branch changes to be reflected in general. In this case here, let me quickly trigger the rebuild so we can ensure it works ok.

Releases are always built as soon as possible.

**Reply by finestructure** [reply]
There's still something wrong, I'm investigating.

**Reply by finestructure** [reply]
Another fix, I missed this when I fixed the file first time around :) https://github.com/Blackjacx/Engine/pull/9

**Reply by Blackjacx** [reply]
Okay so this is not my package version but the version of the documentation?

**Reply by finestructure** [reply]
It's the schema version of the yml file. If we need to make a breaking change we'll bump it to 2.

**Reply by Blackjacx** [reply]
Hmm it's weird, [this build](https://swiftpackageindex.com/builds/5D921933-BE6B-4438-A144-C028FF0E97B3) ends with the comment:
```
Resolve Package Graph
** BUILD SUCCEEDED **
Build complete.
========================================
GenerateDocs
========================================
Done.
```

and quits. No docs egnerated, maybe it wasn't triggered yet. I'll wait another 24hrs.

**Reply by finestructure** [reply]
The build to check is 5.6 / macOS (SPM) instead of “Swift 5.6 on iOS” - but it’s also not run for the latest revision yet.

**Reply by finestructure** [reply]
I've re-triggered it and the docs are now available: https://swiftpackageindex.com/Blackjacx/Engine/develop/documentation/engine

**Reply by Blackjacx** [reply]
WOW, that's awesome. I love it. Thanks a lot. I owe you a beer 🍺
